File: frontend/src/app/services/appointment/appointment.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../auth/services/storage/storage.service';

const URL = 'http://localhost:8080/api/v1/appointment';

@Injectable({
  providedIn: 'root'
})
export class AppointmentService {

  constructor(
    private http: HttpClient,
  ) { }

  getAllAppointment(): Observable<any[]> {
    return this.http.get<[]>(URL, {withCredentials: true});
  }

  getByHealthCode(healthCode: string): Observable<any[]> {
    return this.http.get<[]>(URL + `/by-health-code?healthCode=${healthCode}`, {withCredentials: true});
  }

  getAllDiagnosisById(appointmentId: number): Observable<any[]> {
    return this.http.get<[]>(URL + `/${appointmentId}/diagnosis`);
  }

  addAppointment(patientId: string, appointment: any) {
    return this.http.post(URL + `/by-patient?patientId=${patientId}`, appointment);
  }
}

File: healthcare-clinic-backend/appointment-healthcare-service/src/main/java/com/example/healthcareservice/service/AppointmentService.java
package com.example.healthcareservice.service;

import com.example.healthcareservice.model.Appointment;
import com.example.responsehandling.payload.response.AppResponse;

import java.util.List;

public interface AppointmentService {

    List<Appointment> getAllAppointment();

    List<Appointment> getByHealthCode(String healthCode);

    List<Appointment> getByPatientId(String patientId);

    List<Appointment> getByDoctorId(String doctorId);

    AppResponse addAppointment(String patientId, Appointment appointment);
}
